### PR TITLE
Update Dockerfile

### PR DESCRIPTION
**Pull request by gaohao95**
This PR:
- update MOFED driver to 5.1-2.5.8.0
- compile UCX from source, as there is no UCX binary shipped with this particular MOFED version
- install cuDF from conda instead of compiling from source, since https://github.com/rapidsai/cudf/pull/6726 is merged.

### CODE REVIEW

**Comment by esoha-nvidia** [comment]
The name of the commit could be more descriptive.  I can see that it's "Update Dockerfile" from just looking at the diff.

https://chris.beams.io/posts/git-commit/

Also, it sounds like a single commit is doing 3 different things.  It might be better to have 3 different commits.  That way if one of them doesn't work then we could revert just that one, for example.  The three commits could still be in a single PR if you want, or separate.

Thanks for doing your development on a fork instead of on `rapidsai`!

File: Dockerfile [hunk 1]
```diff
@@ -11,32 +11,29 @@ ENV PATH=${PATH}:/conda/bin
...
+# Setup cuDF and NCCL
+WORKDIR /root
+RUN conda create --name cudf_release \
+    && source activate cudf_release \
+    && conda install -c rapidsai-nightly -c nvidia -c conda-forge -c defaults \
+        cudf=0.17 \
+        python=3.8 \
+        cudatoolkit=11.0 \
+        nccl \
+    && conda clean --all
```
Comment (esoha-nvidia) [inline Dockerfile:23]: Previously you had `-y`.  Now you have removed it.

I tested this command without `-y` and it was interactive.  So I expect that this Dockerfile won't work.

Did you test this?

**Review by esoha-nvidia — COMMENTED** [review]
Thanks for writing this!

How was it tested?

File: Dockerfile (hunk 1 above)
Comment (gaohao95) [reply Dockerfile]: Resovled in https://github.com/rapidsai/distributed-join/pull/48/commits/3e5a77979244758264783aebbf4cb3cba500c562.

**Pull request merged**